### PR TITLE
Check dependencies with dependabot weekly

### PR DESCRIPTION
**Pull request by jlosito**
Similarly to some other hashicorp projects, this adds dependabot so that it check dependencies for updates on a daily basis.

https://github.com/hashicorp/nomad/blob/main/.github/dependabot.yml
https://github.com/hashicorp/packer/blob/main/.github/dependabot.yml

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
[the remark this answers is not in the document]
Comment (johanbrandhorst) [reply .github/dependabot.yml:11]: Also maybe make the dependabot PRs show up on Sundays similar to how nomad does it?

**Review by johanbrandhorst — APPROVED** [review]
Thanks for the updates! We're going to discuss briefly internally whether we want this right now, I'll get back to you!

**Comment by jlosito** [comment]
You might want to limit how many pull requests come in at a given time too, considering that's a lot of dependencies for all of the submodules combined.

**Comment by johanbrandhorst** [comment]
According to https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#open-pull-requests-limit the default limit is 5, which is probably okay.

**Comment by johanbrandhorst** [comment]
There doesn't seem to be a huge amount of interest in this internally right now, so I think I'm going to close this. We will review this at a later time. Thank you for your contribution!

**Pull request closed without merging**